src/Perspex.SceneGraph/Rect.cs

```diff
@@ -297,7 +297,8 @@ public override bool Equals(object obj)
         {
             if (obj is Rect)
             {
-                return this == (Rect)obj;
+                var other = (Rect)obj;
+                return Position == other.Position && Size == other.Size;
```

Comment:
What is the reason for this change? `operator ==` just does `Position == other.Position && Size == other.Size` so it looks like it's just doing the same thing.